### PR TITLE
[FIX] web: fix title overflow

### PR DESCRIPTION
**Pull request by nouraellm**
Steps to reproduce:

- Install stock, purchase
- Create a product with a long name
- Create a purchase order with that product
- Go to portal and access the purchase order

Current behavior:
- Product title overflows in `product.template` form
- Product title overflows when accessing the PO from the portal

Behavior after the PR:
- To fix the problem on the form we add `overflow-wrap` to titles
- To fix the problem in the Portal we add `table-responsive` to the table

opw-3133407

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by fdardenne — COMMENTED** [review]
Hello @nouraellm thanks for the PR,
I think that the titles length problem in the form view is currently discussed by product owners. For me it would rather be an IMP for the current version if the strategy is accepted. 
Can we have your input on this @bouvyd ?

**Comment by bouvyd** [comment]
@fdardenne you're thinking about the title (h1) names on e.g. tasks, etc. which is indeed planned for future versions. The portal stuff is independent.

@nouraellm I'm not sure how this fix relates to the issue in the ticket (in particular, the part about the break-word for oe_title > h1?). Indeed break-word could be useful if you have a really, really long name without any whitespace (e.G. something in german or finnish I guess). Fine by me, but word-break: break-word is deprecated -> https://developer.mozilla.org/en-US/docs/Web/CSS/word-break
> Note: While word-break: break-word is deprecated, it has the same effect, when specified, as word-break: normal and overflow-wrap: anywhere — regardless of the actual value of the [overflow-wrap](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap) property.

=> do that instead please

For the portal stuff, your fix has a rather wide effect since it means that the size of table columns is entirely decided by the first row instead of the browser taking the whole content into account (which is usually preferrable).
|  Before | After  |
|---|---|
|  image |  image |

Sicne super long, non-breakable descriptions are the exception and not the rule, I'd prefer if you added the `table-responsive` class on the **parent** of the portal table instead (in the sale and purchase portals): it will make it possible to scroll for those weird cases, without affecting the layout most of the time.

**Review by fdardenne — COMMENTED** [review]
thank you for the changes !
However it seems that the table does not take the full width with the changes
image

**Comment by nouraellm** [comment]
@fdardenne thanks for the review. Sorry I was not aware of the change, it seems that tables should be wrapped in a div with `table-responsive` so it takes full-width https://getbootstrap.com/docs/3.4/css/#tables-responsive
that's what I did and for the 187 lines changed it is just for nesting the elements within the div

**Review by fdardenne — APPROVED** [review]
LGTM 👍

File: addons/web/static/src/scss/form_view.scss
Comment (rfr-odoo) [inline addons/web/static/src/scss/form_view.scss:479]: There is a particular case where this rule is needed as it's the default value and it doesn't seem to be set before ?

```suggestion
```

Comment (nouraellm) [reply addons/web/static/src/scss/form_view.scss]: thx! should be good now ^^

**Review by reth-odoo — APPROVED** [review]
Doesn't seem to have anything to do with mail, so fine by me :D

**Review by Feyensv — COMMENTED** [review]
Please avoid reindenting the whole file in stable, so that we can clearly see the lines you modify :confused:.

**Comment by Feyensv** [comment]
Still a lot of noise, it might cause conflicts during fw-ports and makes us lose the blame of all those lines :c

**Comment by nouraellm** [comment]
@Feyensv Thanks for your input. I still don't think it is an enough reason to push unindented code, maybe a second opinion can be a plus in such cases ;)

**Comment by adr-odoo** [comment]
We shoudn't block this PR for this. "Move fast" and go ahead.

**Comment by adr-odoo** [comment]
We have an option to hide whitespaces in Github and we can easly see previous commits history in git... The only issue in this case is the forward port and potental conflitcs but I let the dev to handle it.

**Comment by tde-banana-odoo** [comment]
> We shoudn't block this PR for this. "Move fast" and go ahead.

Technically this should target 16 . As there is no traceback in 14 = nothing to do (they should receive as few changes as possible).

Just propagating our master's voice !

**Comment by adr-odoo** [comment]
@nouraellm What if we use ´overflow-wrap: anywhere;´ instead of ´table-responsive´? Plus, it will avoid to reindent the existing code.

**Comment by bouvyd** [comment]
@nouraellm your fix works, both in portal and in the reports so let's move along and close this debate :)

@robodoo r+

**Pull request closed without merging**
